webhelper/src/gwh-browser.c

```diff
@@ -136,14 +136,15 @@ static void       inspector_set_detached      (GwhBrowser *self,
 
 static void
 set_location_icon (GwhBrowser  *self,
-                   const gchar *icon_uri)
+                   cairo_surface_t *favicon)
 {
   gboolean success = FALSE;
   
-  if (icon_uri) {
+  if (favicon) {
     GdkPixbuf *icon;
-    
-    icon = gwh_pixbuf_new_from_uri (icon_uri, NULL);
+    GtkImage *image;
+    gtk_image_set_from_surface (image, favicon);
+    icon = gtk_image_get_pixbuf(image);
```

Comment:
does this work?  I would be highly surprised if it does, GtkImage is not necessarily (well, I think never) using Pixbuf backed data, and so the `get_pixbuf()` method will probably return `NULL` here, doesn't it?